src/overlays/actors/ovl_En_Part/z_en_part.c

```diff
@@ -21,23 +26,299 @@ const ActorInit En_Part_InitVars = {
+            this->actor.velocity.y = Math_Rand_ZeroOne() * 3.0f + 8.0f;
+            this->actor.gravity = -0.6f - Math_Rand_ZeroOne() * 0.3f;
+            this->rotZSpeed = 0.15f;
+            break;
+        case 5:
+        case 6:
+        case 7:
+        case 8:
+            this->actor.posRot.rot.y = this->actor.parent->shape.rot.y;
+            if (this->displayList == D_06015380) {
+                sign = -1.0f;
+            }
+            this->actor.velocity.y = 0.0f;
+            this->actor.speedXZ = 6.0f * sign;
+            this->actor.gravity = -1.2f;
+            this->rotZSpeed = 0.15f * sign;
+            ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawFunc_Circle, 30.0f);
+            this->timer = 18;
+            break;
+    }
+}
+
+void func_80ACE13C(EnPart* this, GlobalContext* globalCtx) {
+    s32 i;
+    Vec3f pos;
+    Vec3f velocity = { 0.0f, 0.0f, 0.0f };
+    Vec3f accel = { 0.0f, 0.0f, 0.0f };
+    Vec3f zero = { 0.0f, 0.0f, 0.0f };
+
+    if ((this->actor.params == 12) || (this->actor.params == 13)) {
+        func_8002E4B4(globalCtx, &this->actor, 5.0f, 15.0f, 0.0f, 0x1D);
+
+        if ((this->actor.bgCheckFlags & 1) || (this->actor.posRot.pos.y <= this->actor.groundY)) {
+            this->action = 4;
+            this->actor.speedXZ = 0.0f;
+            this->actor.gravity = 0.0f;
+            this->actor.velocity.y = 0.0f;
+        }
+
+        if ((this->actor.params == 13) && (this->actor.parent != NULL) && (this->actor.parent->update == NULL)) {
+            this->actor.parent = NULL;
+        }
+    } else if (this->timer <= 0) {
+        switch (this->actor.params) {
+            case 1:
+            case 9:
+            case 10:
+            case 14:
+                EffectSsDeadDb_Spawn(globalCtx, &this->actor.posRot.pos, &zero, &zero,
+                                     (s16)(this->actor.scale.y * 100.0f) * 40, 7, 255, 255, 255, 255, 0, 255, 0, 1, 9,
+                                     true);
+                break;
+            case 3:
+            case 11:
+                EffectSsDeadDb_Spawn(globalCtx, &this->actor.posRot.pos, &zero, &zero,
+                                     (s16)(this->actor.scale.y * 100.0f) * 40, 7, 255, 255, 255, 255, 0, 0, 255, 1, 9,
+                                     true);
+                break;
+            case 4:
+                for (i = 7; i >= 0; i--) {
+                    pos.x = this->actor.posRot.pos.x + Math_Rand_CenteredFloat(60.0f);
+                    pos.y = this->actor.posRot.pos.y + this->actor.shape.unk_08 * this->actor.scale.y +
+                            Math_Rand_CenteredFloat(50.0f);
+                    pos.z = this->actor.posRot.pos.z + Math_Rand_CenteredFloat(60.0f);
+                    velocity.y = Math_Rand_ZeroOne() + 1.0f;
+                    EffectSsDtBubble_SpawnColorProfile(globalCtx, &pos, &velocity, &accel, Math_Rand_S16Offset(80, 100),
+                                                       25, 0, 1);
```

Comment:
randXZ (the last arg there) is a bool so you can make that `true`
I also prefer to call `zero` `zeroVec` but I leave that up to you